Update `api` references in Image, scripts and tests, where `api2` seems intended

nimmolo opened the pull request:
There are a few places in the code that only access one API, and i feel like they should probably be using API2 at this point.

Please check me on this!

JoeCohen commented on script/make_eol_xml:
```diff
@@ -84,8 +84,8 @@ end
 if testing
-  LOCAL_URL = "https://localhost:3000/api/observations?id=1"
-  REMOTE_URL = "https://mushroomobserver.org/api/observations?id=1"
+  LOCAL_URL = "https://localhost:3000/api2/observations?id=1"
```
Should this (and line 90) be `http:`?

JoeCohen replied:
1. We can't go to `https://localhost...`. So things would blow up if someone tried that. 
2. I don't know if we are really using the eol code currently. @mo-nathan is the best person to answer that.

pellaea replied:
Agreed, the local server on the dev VM isn't secure.

nimmolo replied:
Changed to `http://localhost...` in both places

JoeCohen commented on script/old/refresh_controllers:
```diff
@@ -186,7 +186,7 @@ end
 norobots = [
   "/ajax/",
-  "/api/"
+  "/api2/"
```
@pellaea Are the scripts in `script/old` used?
If yes, what's intended here (`api` vs `api2`)?

JoeCohen commented on script/old/upload_observation.rb:
```diff
@@ -515,7 +515,7 @@ def post_data
   def post_observation
     http = Net::HTTP.new(HOST, PORT)
-    path = make_path("/api/observations",
+    path = make_path("/api2/observations",
```
@pellaea Are the scripts in `script/old` used?
If yes, what's intended here (`api` vs `api2`)?

pellaea commented:
It's probably time to remove the old scripts.  I'd initially left them in there in case they were still used, but after a few years and I haven't used a single one in all that time, I'd say it's safe to remove them!  Same with test_api.rb.  Totally unnecessary now.  (And might not even work...)

JoeCohen commented on test/models/localization_files_test.rb:
```diff
@@ -144,7 +144,7 @@ def source_files(*paths, &block)
   def test_api_error_translations
     tags = []
-    file = "#{::Rails.root}/app/classes/api/errors.rb"
+    file = "#{::Rails.root}/app/classes/api2/errors.rb"
```
@pellaea: Should we run this for both `.../api/errors` and `.../api2/errors? 
If not, this looks right.
(And I think it explains something that puzzled me: How did we end up with the missing translation that @nimmolo fixed in #995?)

nimmolo replied:
Ahaa!

JoeCohen commented on test/models/localization_files_test.rb:
```diff
@@ -154,7 +154,7 @@ def test_api_error_translations
       end
     end
-    Dir.glob("#{::Rails.root}/app/classes/api/parsers/*.rb").each do |file|
+    Dir.glob("#{::Rails.root}/app/classes/api2/parsers/*.rb").each do |file|
```
See above question.

nimmolo replied:
I'd in favor of testing api2 only, because i'm hoping we will see fit to delete API very soon and it's less busywork. But I can add it now, merge, and add the delete to the "Sunset the API" branch if you like.

nimmolo replied:
Oh also - I believe they use the same localization files, so testing one should be enough.

pellaea commented:
We definitely should not be doing any development on API!  Even if we were
to keep it, we should not be adding tests or fixing bugs or anything on
it.  I hope we can finally clear this up at the meeting tomorrow.  You both
seem to think that keeping API around is some big burden.  It should not be.

On Fri, Jun 10, 2022 at 4:46 PM andrew nimmo ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In test/models/localization_files_test.rb
> <https://github.com/MushroomObserver/mushroom-observer/pull/1000#discussion_r894877769>
> :
>
> > @@ -154,7 +154,7 @@ def test_api_error_translations
>          tags << "api_#{Regexp.last_match(1).underscore.tr("/", "_")}".to_sym
>        end
>      end
> -    Dir.glob("#{::Rails.root}/app/classes/api/parsers/*.rb").each do |file|
> +    Dir.glob("#{::Rails.root}/app/classes/api2/parsers/*.rb").each do |file|
>
> Oh also - I believe they use the same localization files, so testing one
> should be enough.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MushroomObserver/mushroom-observer/pull/1000#discussion_r894877769>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYTNNNJLOHPPRGUDBJKTH3VOOSSHANCNFSM5YOU3MJA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>

nimmolo commented:
Keeping the API frozen is a blocker for zeitwerk, so yea it is kind of a burden.

pellaea commented:
Right, which is the reason we're talking about retiring it now.  But why in
the world would we be considering adding tests for it in the meantime?  Or
ever?

On Fri, Jun 10, 2022 at 5:04 PM andrew nimmo ***@***.***>
wrote:

> Keeping the API frozen is a blocker for zeitwerk, so yea it is kind of a
> burden.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MushroomObserver/mushroom-observer/pull/1000#issuecomment-1152737359>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYTNNLK6Y3QL5ZCGB5UATDVOOUVVANCNFSM5YOU3MJA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

nimmolo commented:
Right! Well this PR effectively _subtracts_ a test from the API, I think is why Joe raised this issue. However, the same translations are still getting tested.

Incidentally, I updated localhost to use http, and deleted the old scripts and test_api.rb. I'm basically doing these small PRs, where I'm sure of relatively minor changes, to try and remove the noise and cruft from the bigger ones. Let me know if you feel like it's mergeable, both of yous! :)

pellaea commented:
Yes, I'm totally down with merging the deletion of old scripts and
test_api.rb.

On Fri, Jun 10, 2022 at 5:11 PM andrew nimmo ***@***.***>
wrote:

> Right! Well this PR effectively *subtracts* a test from the API, I think
> is why Joe raised this issue. However, the same translations are still
> getting tested.
>
> Incidentally, I updated localhost to use http, and deleted the old scripts
> and test_api.rb. I'm basically doing these small PRs, where I'm sure of
> relatively minor changes, to try and remove the noise and cruft from the
> bigger ones. Let me know if you feel like it's mergeable, both of yous! :)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MushroomObserver/mushroom-observer/pull/1000#issuecomment-1152741242>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYTNNL6RBH5DPQFUDVV3Q3VOOVPRANCNFSM5YOU3MJA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

JoeCohen commented:
I'm fine with merging it.
Because TestApi is now toast, it makes sense to also revise `test_api_error_translations`.
NOTE: If we choose to retain the old API, do we lose something by the above changes?
In particular, could a change other than in API cause it to break? And could that breakage be detected by TestApi?

pellaea commented:
I don't think we lose anything.

On Fri, Jun 10, 2022 at 5:23 PM Joseph D. Cohen ***@***.***>
wrote:

> I'm fine with merging it.
> Because TestApi is now toast, it makes sense to also revise
> test_api_error_translations.
> NOTE: If we choose to retain the old API, do we lose something by the
> above changes?
> In particular, could a change other than in API cause it to break? And
> could that breakage be detected by TestApi?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/MushroomObserver/mushroom-observer/pull/1000#issuecomment-1152747674>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAYTNNK7I7SZC4JJ45QUZKTVOOW3LANCNFSM5YOU3MJA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
